src/dotenv.ts

```diff
@@ -53,7 +53,7 @@ export async function setupDotenv(options: DotenvOptions): Promise<Env> {
 
   // Fill process.env
   for (const key in environment) {
-    if (!key.startsWith("_") && targetEnvironment[key] === undefined) {
+    if (!key.startsWith("_")) {
```

Comment:
we already perform this check (to avoid overriding process.env from .env) on L76, and by repeating it here we can't ever update values

Reply:
This would be a behavior change. The idea is that dotenv acts as (defaults) and we always respect actual os env variables with highest priority.

What if, instead of flagging the entire `process.env` as "dotenv applied", we flag values set by dot env individually in the first run and in second runs, we only apply values previously were also applied by dotenv alone.

(with this changes, we can remove side-effect flag in loadDotEnv)

Reply:
I think even at the moment, we do still respect actual os env variables, because we already ran `Object.assign(environment, options.env);` which means that `environment[key] === process.env[key]`.

This only differs in behaviour when that key is updated.

Reply:
Yes, but with other change, it would be one time (updates to `process.env` will be lost because we flag it)